File: app/src/main/java/com/mamba/picme/features/idphoto/components/IdPhotoTabRow.kt
package com.mamba.picme.features.idphoto.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import com.mamba.picme.R
import com.mamba.picme.features.idphoto.IdPhotoTab

/** 底部 4-tab 行：底色/尺寸/边缘/修补。 */
@Composable
fun IdPhotoTabRow(selected: IdPhotoTab, onSelect: (IdPhotoTab) -> Unit) {
    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceEvenly) {
        IdPhotoTab.entries.forEach { tab ->
            val labelRes = when (tab) {
                IdPhotoTab.BG_COLOR -> R.string.id_photo_tab_color
                IdPhotoTab.SIZE -> R.string.id_photo_tab_size
                IdPhotoTab.EDGE -> R.string.id_photo_tab_edge
                IdPhotoTab.REPAIR -> R.string.id_photo_tab_repair
            }
            FilterChip(
                selected = tab == selected,
                onClick = { onSelect(tab) },
                label = {
                    Text(
                        stringResource(labelRes),
                        color = if (tab == selected) MaterialTheme.colorScheme.onPrimary else Color.White
                    )
                },
                colors = FilterChipDefaults.filterChipColors(
                    containerColor = Color(0xFF2A2A2A),
                    selectedContainerColor = MaterialTheme.colorScheme.primary
                )
            )
        }
    }
}

File: iosApp/PoLang/Features/IdPhoto/Components/IdPhotoTabRow.swift
import SwiftUI

/// 证照屏底部 Tab 行（对标 specs/screens/idphoto.yaml §2.3；定稿截图 tmp/ui-reference/idphoto-02-screen.png）。
/// 底色/尺寸/边缘/修补 四个 filter chip spaceEvenly 平铺：
/// 选中 = colorScheme.primary(dark) 容器 + onPrimary 字；未选中 = idphoto.tabUnselectedContainer + 白字。
struct IdPhotoTabRow: View {
    let selected: IdPhotoTab
    let onSelect: (IdPhotoTab) -> Void

    var body: some View {
        HStack(spacing: 0) {
            ForEach(IdPhotoTab.allCases) { tab in
                chip(tab)
                    .frame(maxWidth: .infinity)   // 等宽槽位 ≈ Arrangement.SpaceEvenly
            }
        }
        .frame(maxWidth: .infinity)
    }

    private func chip(_ tab: IdPhotoTab) -> some View {
        let isSelected = tab == selected
        let scheme = AppColorScheme.dark
        return Button {
            onSelect(tab)
        } label: {
            Text(L(tab.labelKey))
                .font(AppTypography.labelLarge.font)
                .foregroundStyle(isSelected ? scheme.onPrimary : .white)
                .padding(.horizontal, Spacing.lg)
                .frame(height: ChipTokens.smallHeight)
                .background(
                    RoundedRectangle(cornerRadius: AppRadius.small)
                        .fill(isSelected ? scheme.primary : IdPhotoTokens.tabUnselectedContainer)
                )
        }
        .buttonStyle(.plain)
        .accessibilityAddTraits(isSelected ? [.isSelected] : [])
    }
}

private extension IdPhotoTab {
    /// Tab 标题键（§11；域类型 IdPhotoTab 无 labelKey——UI 侧映射，顺序契约 bgColor/size/edge/repair）
    var labelKey: String {
        switch self {
        case .bgColor: return "id_photo_tab_color"
        case .size: return "id_photo_tab_size"
        case .edge: return "id_photo_tab_edge"
        case .repair: return "id_photo_tab_repair"
        }
    }
}
